crates/polars-plan/src/plans/optimizer/predicate_pushdown/utils.rs

```diff
@@ -240,6 +232,7 @@ pub enum PushdownEligibility {
 #[allow(clippy::type_complexity)]
 pub fn pushdown_eligibility(
     projection_nodes: &[ExprIR],
+    new_predicates: &[(Arc<str>, ExprIR)],
```

Comment:
drive-by perf, we only need to check the newly added predicates, currently we check the entire list of predicates at every LP we pass through 🫠.

I seem to remember a while ago someone complained about `collect()` speed regression, maybe this was the cause 😅

Reply:
Nice!